tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1120 @@
+
+        @steps: List all smart variables associated to puppet class by puppet
+        class name.
+
+        @assert: Smart Variables listed for specific puppet class by puppet
+        class name.
+
+        @CaseLevel: Integration
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_list_variables_by_puppetclass_id(self):
+        """List all smart variables associated to puppet class by puppet class
+        id.
+
+        @id: ba261c80-699c-4b75-9e07-b80cb0a80c55
+
+        @steps: List all smart variables associated to puppet class by puppet
+        class id.
+
+        @assert: Smart Variables listed for specific puppet class by puppet
+        class id.
+
+        @CaseLevel: Integration
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create(self):
+        """Create a Smart Variable.
+
+        @id: 4a8589bf-7b11-48e8-a25d-984bea2ba676
+
+        @steps:
+
+        1. Create a smart Variable with Valid name and default value.
+
+        @assert: The smart Variable is created successfully.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create(self):
+        """Create Smart Variable with invalid name.
+
+        @id: 09a67cb6-5de0-41b3-90f4-593323936c6c
+
+        @steps:
+
+        1. Create a smart Variable with Invalid name and valid default value.
+
+        @assert: The smart Variable is not created.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_delete_smart_variable_by_id(self):
+        """Delete a Smart Variable by id.
+
+        @id: 19fedbdf-48a1-46a7-b184-615a0efd7b4e
+
+        @steps:
+
+        1. Delete a smart Variable by id.
+
+        @assert: The smart Variable is deleted successfully.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_delete_smart_variable_by_name(self):
+        """Delete a Smart Variable by name.
+
+        @id: 19fedbdf-48a1-46a7-b184-615a0efd7b4e
+
+        @steps:
+
+        1. Delete a smart Variable by name.
+
+        @assert: The smart Variable is deleted successfully.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_update_variable_puppet_class(self):
+        """Update Smart Variable's puppet class.
+
+        @id: 6c3e2da9-420c-4e39-8b71-e5be6b605bd7
+
+        @steps:
+
+        1. Create a smart variable with valid name and default value.
+        2. Add this variable to some host/hostgroup.
+        3. Update the puppet class associated to the smart variable created in
+        step1.
+
+        @assert:
+
+        1. The variable is updated with new puppet class.
+        2. In Host/HostGroup -> variables tab, the smart variable is updated
```

Comment:
`In Host/HostGroup -> variables tab, the smart variable is updated`
There are no tab term in CLI